### PR TITLE
add Library.whoCanComment and LibraryPermission.ADD_COMMENTS

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @leogrim @andrewconner

### CODE REVIEW

**Comment by camhashemi** [comment]
just adding model updates, no enforcement.

File: kifi-backend/modules/common/app/com/keepit/model/LibraryInvite.scala [hunk 1]
```diff
@@ -108,6 +108,24 @@ object LibraryInvitePermissions {
...
 
+sealed abstract class LibraryCommentPermissions(val value: String)
+object LibraryCommentPermissions {
+  case object COLLABORATOR extends LibraryCommentPermissions("collaborator") // collaborators and if Library.organizationMemberAccess == READ_WRITE, org-members
+  case object ANYONE extends LibraryCommentPermissions("anyone") // anyone with view permissions
+
+  implicit val format: Format[LibraryCommentPermissions] = Format(
+    Reads { j => j.validate[String].map(LibraryCommentPermissions(_)) },
+    Writes { o => JsString(o.value) }
+  )
```
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/model/LibraryInvite.scala]: Make this a `EnumFormat.safeReads`?

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/PermissionCommander.scala [hunk 2]
```diff
@@ -146,27 +146,35 @@ class PermissionCommanderImpl @Inject() (
       } getOrElse libPermissions
     }
 
+    def hasOrgWriteAccess(library: Library): Boolean = {
+      library.organizationMemberAccess.contains(LibraryAccess.READ_WRITE) &&
+        library.organizationId.exists(orgMembershipsById(_).isDefined)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/PermissionCommander.scala]: `library.organizationId.flatMap(orgMembershipsById.get).isDefined`

File: kifi-backend/modules/shoebox/test/com/keepit/commanders/PermissionCommanderTest.scala [hunk 3]
```diff
@@ -134,6 +137,20 @@ class PermissionCommanderTest extends TestKitSupport with SpecificationLike with
...
       }
+      "add comment permissions" in {
+        withDb(modules: _*) { implicit injector =>
+          db.readWrite { implicit session =>
+            val (owner, collab, follower, rando, noone, publicLib, secretLib) = setupLibs()
+            val all = Set(owner, collab, follower, rando, noone)
+
+            Set(publicLib, secretLib).foreach { lib =>
+              val whoCanKeep = all.filter { x => permissionCommander.getLibraryPermissions(publicLib.id.get, x).contains(LibraryPermission.ADD_COMMENTS) }
+              whoCanKeep === Set(owner, collab)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/test/com/keepit/commanders/PermissionCommanderTest.scala]: `whoCanComment ===`

Comment (camhashemi) [inline kifi-backend/modules/shoebox/test/com/keepit/commanders/PermissionCommanderTest.scala]: :8ball:

**Comment by leogrim** [comment]
Looks reasonable overall

**Comment by ryanpbrewster** [comment]
lgtm as well. I agree with everything you said in-person and think we should think more about how to establish a good convention moving forward.

File: kifi-backend/modules/shoebox/marketing [hunk 4]
```diff
@@ -1 +1 @@
-Subproject commit 5c8a4af4be066b7407e10bdae487282c02072a67
+Subproject commit 8f77efab4b446e3adecfc07ac56f297686c42a85
```
Comment (camhashemi) [inline kifi-backend/modules/shoebox/marketing:1]: up to date

Comment (camhashemi) [inline kifi-backend/modules/shoebox/marketing:1]: for now

**Pull request merged**